src/components/HAAPIProcessor.js

```diff
@@ -97,10 +99,18 @@ export default function HAAPIProcessor(props) {
                     inputProblem={step.inputProblem}
                 />
             case 'views/select-authenticator/index':
-                return <Selector
+                if (config.authenticator == null) {
+                    return <Selector
                         actions={haapiResponse.actions}
                         submitForm={(url, method) => submitForm( null, url, method)}
-                        />
+                    />
+                } else {
+                    return <PreSelectedSelector
```

Comment:
Why have this component at all? In the future all clients will have multiple authenticator. So instead, for old client, this can be worked around in tests (selecting the only option)

Reply:
It's hard to say if the social/passkeys authenticator will be tested by these tests. AFAIK the QA team is working on integrating the LIE FE tests with Curity, these authenticators might be tested by using a real frontend web app.

I am also not so sure how the frontend client will implement choosing the authenticator from multiple ones, they might react to a user action and then start the Curity flow and choose the authenticator without user interaction, so the behavior will be similar to the `PreSelectedAuthenticator`. 

Moreover, the `PreSelectedAuthenticator` component behaves similarly to the frontend clients will behave with the use of FT. 

I understand your point, but I think that we can keep this component and implement solution for choosing between multiple authenticators, when we will need it :)